src/dimr_bmi.c

```diff
@@ -5,65 +5,47 @@
 #include <string.h>
 #include <assert.h>
 
-#include "sealock.h"
+#include "zsf_config.h"
 
 #define ZSF_VERBOSE 1
-#define ZSF_MAX_LOCKS 50
 #define ZSF_KEY_SEPARATOR '/'
-
 #define ZSF_TO_DIMR_STATUS(s) ((s) == 0 ? DIMR_BMI_OK : DIMR_BMI_FAILURE)
 
 
-// Keep track of multiple sea-lock instances.
-sealock_state_t locks[ZSF_MAX_LOCKS];
-double start_time = 0;
-double current_time = 0;
-double end_time = 0;
+// Global conbfiguration.
+zsf_config_t config;
```

Comment:
This makes me sad. Do I understand that our implementation of BMI requires global data? I see that BMI 2.0 allows you to store the data in the BMI struct https://github.com/csdms/bmi-c/blob/master/bmi.h

Reply:
I feel your pain, but BMI 1.0 doesn't have such a thing, or at the very least a shared BMI object isn't implemented in 'our version of BMI'. I would prefer to have the state stored outside the zsf dll. We currently probably have issues with thread safety in parallel runs.

Reply:
Since there is not other way to solve issues with having a global struct (or pointer to a struct) I won't fix this in the current issue.